File: ruby/016.rb
#
# Project Euler: Challenge 003 - http://projecteuler.net/problem=3
#
# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143 ?
#
# Answer: 6857

p (2**1000).to_s.split(//).map(&:to_i).inject(0) {|r, n| r+n }

File: haskell/016.hs
--
-- Project Euler: Challenge 016 - http://projecteuler.net/problem=16
--
-- 215 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
--
-- What is the sum of the digits of the number 21000?
--
-- Answer: 1366
--
import Euler(digits')

main = do print . sum $ digits' $ 2^1000
